diff --git a/package/lean/luci-app-cpufreq/root/etc/config/cpufreq b/package/lean/luci-app-cpufreq/root/etc/config/cpufreq
index 0e7eaede3..f7128226f 100644
--- a/package/lean/luci-app-cpufreq/root/etc/config/cpufreq
+++ b/package/lean/luci-app-cpufreq/root/etc/config/cpufreq
@@ -1,8 +1,8 @@
 
 config settings 'cpufreq'
-	option governor ''
-	option minfreq ''
-	option maxfreq ''
+	option governor 'schedutil'
+	option minfreq '600000'
+	option maxfreq '1512000'
 	option upthreshold '50'
 	option factor '10'
 
diff --git a/package/lean/luci-app-turboacc/Makefile b/package/lean/luci-app-turboacc/Makefile
index f61f4f78e..f1de50f17 100644
--- a/package/lean/luci-app-turboacc/Makefile
+++ b/package/lean/luci-app-turboacc/Makefile
@@ -22,7 +22,7 @@ LUCI_PKGARCH:=all
 define Package/$(PKG_NAME)/config
 config PACKAGE_$(PKG_NAME)_INCLUDE_shortcut-fe
 	bool "Include Shortcut-FE"
-	depends on PACKAGE_$(PKG_NAME)_INCLUDE_flow-offload=n
+	#depends on PACKAGE_$(PKG_NAME)_INCLUDE_flow-offload=n
 	default y if LINUX_4_9
 	default n
 
diff --git a/package/lean/luci-app-turboacc/root/etc/config/turboacc b/package/lean/luci-app-turboacc/root/etc/config/turboacc
index 067aad2763..cf30d7359d 100644
--- a/package/lean/luci-app-turboacc/root/etc/config/turboacc
+++ b/package/lean/luci-app-turboacc/root/etc/config/turboacc
@@ -2,8 +2,8 @@
 config turboacc 'config'
 	option sw_flow '1'
 	option hw_flow '1'
-	option sfe_flow '1'
-	option sfe_bridge '1'
+	option sfe_flow '0'
+	option sfe_bridge '0'
 	option sfe_ipv6 '0'
 	option bbr_cca '1'
 	option fullcone_nat '1'
diff --git a/target/linux/rockchip/config-default b/target/linux/rockchip/config-default
index 1b2d5e1903..47c1193432 100644
--- a/target/linux/rockchip/config-default
+++ b/target/linux/rockchip/config-default
@@ -79,6 +79,7 @@ CONFIG_ARCH_WANT_HUGE_PMD_SHARE=y
 CONFIG_ARM64=y
 CONFIG_ARM64_4K_PAGES=y
 CONFIG_ARM64_CONT_SHIFT=4
+CONFIG_ARM64_CRYPTO=y
 CONFIG_ARM64_ERRATUM_1165522=y
 CONFIG_ARM64_ERRATUM_1286807=y
 CONFIG_ARM64_PAGE_SHIFT=12
@@ -107,13 +108,69 @@ CONFIG_COMMON_CLK=y
 CONFIG_CPU_RMAP=y
 CONFIG_CRYPTO_AEAD=y
 CONFIG_CRYPTO_AEAD2=y
+CONFIG_CRYPTO_AES_ARM64=y
+CONFIG_CRYPTO_AES_ARM64_BS=y
+CONFIG_CRYPTO_AES_ARM64_CE=y
+CONFIG_CRYPTO_AES_ARM64_CE_BLK=y
+CONFIG_CRYPTO_AES_ARM64_CE_CCM=y
+CONFIG_CRYPTO_AES_ARM64_NEON_BLK=y
+CONFIG_CRYPTO_AUTHENC=y
+CONFIG_CRYPTO_CBC=y
+CONFIG_CRYPTO_CHACHA20=y
+CONFIG_CRYPTO_CHACHA20_NEON=y
 CONFIG_CRYPTO_CRC32=y
+CONFIG_CRYPTO_CRYPTD=y
+CRYPTO_CRCT10DIF_ARM64_CE=y
+CONFIG_CRYPTO_CTR=y
+CONFIG_CRYPTO_DEV_CCREE=y
+CONFIG_CRYPTO_DEV_HISI_SEC=y
+CONFIG_CRYPTO_DRBG=y
+CONFIG_CRYPTO_DRBG_CTR=y
+CONFIG_CRYPTO_DRBG_HASH=y
+CONFIG_CRYPTO_DRBG_HMAC=y
+CONFIG_CRYPTO_DRBG_MENU=y
+CONFIG_CRYPTO_ECB=y
+CONFIG_CRYPTO_GF128MUL=y
+CONFIG_CRYPTO_GHASH_ARM64_CE=y
 CONFIG_CRYPTO_HASH=y
 CONFIG_CRYPTO_HASH2=y
+CONFIG_CRYPTO_HMAC=y
+CONFIG_CRYPTO_HW=y
+CONFIG_CRYPTO_JITTERENTROPY=y
+CONFIG_CRYPTO_LIB_DES=y
+CONFIG_CRYPTO_LIB_SHA256=y
 CONFIG_CRYPTO_MANAGER=y
 CONFIG_CRYPTO_MANAGER2=y
+CONFIG_CRYPTO_MD5=y
+CONFIG_CRYPTO_NHPOLY1305=y
+CONFIG_CRYPTO_NHPOLY1305_NEON=y
+CONFIG_CRYPTO_NULL=y
 CONFIG_CRYPTO_NULL2=y
+CONFIG_CRYPTO_POLY1305=y
+CONFIG_CRYPTO_RNG=y
 CONFIG_CRYPTO_RNG2=y
+CONFIG_CRYPTO_RNG_DEFAULT=y
+CONFIG_CRYPTO_SEQIV=y
+CONFIG_CRYPTO_SHA1=y
+CONFIG_CRYPTO_SHA1_ARM64_CE=y
+CONFIG_CRYPTO_SHA256=y
+CONFIG_CRYPTO_SHA256_ARM64=y
+CONFIG_CRYPTO_SHA2_ARM64_CE=y
+CONFIG_CRYPTO_SHA3=y
+CONFIG_CRYPTO_SHA3_ARM64=y
+CONFIG_CRYPTO_SHA512=y
+CONFIG_CRYPTO_SHA512_ARM64=y
+CONFIG_CRYPTO_SHA512_ARM64_CE=y
+CONFIG_CRYPTO_SIMD=y
+CONFIG_CRYPTO_SM3=y
+CONFIG_CRYPTO_SM3_ARM64_CE=y
+CONFIG_CRYPTO_SM4=y
+CONFIG_CRYPTO_SM4_ARM64_CE=y
+CONFIG_CRYPTO_USER_API=y
+CONFIG_CRYPTO_USER_API_AEAD=y
+CONFIG_CRYPTO_USER_API_HASH=y
+CONFIG_CRYPTO_USER_API_RNG=y
+CONFIG_CRYPTO_USER_API_SKCIPHER=y
+CONFIG_CRYPTO_XTS=y
 CONFIG_DCACHE_WORD_ACCESS=y
 CONFIG_DMA_DIRECT_REMAP=y
 CONFIG_DMA_REMAP=y
@@ -280,6 +337,7 @@ CONFIG_RODATA_FULL_DEFAULT_ENABLED=y
 CONFIG_RPS=y
 CONFIG_RWSEM_SPIN_ON_OWNER=y
 CONFIG_SERIAL_8250_FSL=y
+CONFIG_SG_SPLIT=y
 CONFIG_SMP=y
 CONFIG_SPARSEMEM_VMEMMAP_ENABLE=y
 CONFIG_SPARSE_IRQ=y